source/dub/dub.d

```diff
@@ -603,15 +599,33 @@ class Dub {
 						import std.stdio;
 						import core.runtime;
 
-						void main() { writeln("All unit tests have been run successfully."); }
-						shared static this() {
+						// libraries need a main function
+						version (dub_test_main) void main() {}
+
+						bool defaultUnitTester() {
+							import core.stdc.stdlib : exit;
+							foreach (modul; allModules)
+								foreach (test; __traits(getUnitTests, modul))
+									test();
+							writeln("All unit tests have been run successfully.");
+							exit(0);
+							return true;
+						}
+
+						shared static this()
+						{
 							version (Have_tested) {
 								import tested;
 								import core.runtime;
 								import std.exception;
 								Runtime.moduleUnitTester = () => true;
```

Comment:
Again, `() => false` due to the applications `main()` getting called otherwise.

Reply:
`return false` means the program exits with code 2.

This means I need to run the module destructors manually and then exit with code 0.

Reply:
Hm, you are right, that sucks.... I was sure that something like this was mentioned as a way to skip `main` during the last discussion about the unit testing topic. There was also supposed to be some improvement in this area in a recent DMD release, but I could neither find anything in the change log, nor the discussion.

An alternative possibility could be to define a C `main()`, but the question is whether that is less risky overall than skipping the module finalizers. Speaking of which, what about just calling `Runtime.terminate();` just before `exit(0);`?

Reply:
Bit ahead of you there.